tests/library/trace-viewer.spec.ts

```diff
@@ -1914,3 +1915,22 @@ test('should render locator descriptions', async ({ runAndTrace, page }) => {
     - treeitem /Click.*input.*first/
   `);
 });
+
+test('should load trace from HTTP', async ({ showTraceViewer, server }) => {
+  const [traceViewer, res] = await Promise.all([
+    showTraceViewer([server.PREFIX]),
+    new Promise<http.ServerResponse>(resolve => {
+      server.setRoute('/', (req, res) => resolve(res));
+    }),
```

Comment:
```suggestion
    server.waitForRequest(),
```

does this work too?

Reply:
Oh, we have a `waitForRequest`? That's neat! Yes that works, i'll refactor it.